### PR TITLE
Preview of translations

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
Introduces a script to update translations from an extracted Crowdin ZIP folder.

The script makes changes to pyright and stubs projects that need to be committed to those projects.

It currently represents the translations as of 9:24 UTC on the 5th September.

Tag the work-in-progress languages (all of them) as "preview" in the language dialog for now so we can merge this into the beta.

Reference content is also now available though that's via the CMS sync.

## Testing

Basic test table, confirming that UI strings, error messages, typeshed and Reference/Ideas changes show up:

- [x] es-es
- [x] fr
- [x] ja
- [x] ko
- [x] zh-CN
- [x] zh-TW
    - Note display.scroll isn't translated yet which was my test case for the others. But e.g. panic is.

### CODE REVIEW

File: bin/update-translations.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+mkdir -p crowdin/translated
+for language in $languages; do
+    lower="${language,,}"
+    prefix="${1}/${language}/new/apps/python-editor-v3"
+    cp "${prefix}/ui.en.json" "crowdin/translated/ui.${lower}.json"
+    cp "${prefix}/errors.en.json" "../pyright/packages/pyright-internal/src/localization/simplified.nls.${lower}.json"
+    cp "${prefix}/api.en.json" "../micropython-microbit-stubs/crowdin/translated/api.${lower}.json"
+done
+npm run i18n:convert
+npm run i18n:compile || true # Failing for the moment
```
Comment (microbit-matt-hillsdon) [inline bin/update-translations.sh:29]: I fixed this, I think!

```suggestion
npm run i18n:compile
```

File: src/settings/settings.tsx [hunk 2]
```diff
@@ -6,7 +6,6 @@
 import { createContext, ReactNode, useContext } from "react";
 import { useStorage } from "../common/use-storage";
 import { defaultCodeFontSizePt } from "../deployment/misc";
-import { stage } from "../environment";
 
 export interface Language {
```
Comment (microbit-matt-hillsdon) [inline src/settings/settings.tsx:10]: I'll get Emma Smart to review this (visually) and offer a view on order.

**Review by microbit-robert — COMMENTED** [review]
Code changes look good. Typeshed, error messages and UI strings changing on language settings.

File: src/settings/settings.tsx (hunk 2 above)
Comment (microbit-matt-hillsdon) [reply src/settings/settings.tsx:10]: I've reordered by English language name which Emma was happy with. Makes more sense than sorting by a technical id.

**Comment by microbit-matt-hillsdon** [comment]
Crowdin CMS content is now successfully syncing after fixing a variety of tag-related errors.

**Comment by microbit-matt-hillsdon** [comment]
Corresponding Foundation pyright repo and stubs PRs:

Stubs: https://github.com/microbit-foundation/micropython-microbit-stubs/pull/63
Pyright: https://github.com/microbit-foundation/pyright/pull/18

**Review by microbit-robert — APPROVED** [review]
LGTM.

**Pull request merged**